app/models/upload_reference.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class UploadReference < ActiveRecord::Base
+  belongs_to :upload
+  belongs_to :target, polymorphic: true
+
+  def self.exist!(upload_ids: [], target: nil, target_type: nil, target_id: nil)
```

Comment:
Not sure about that name 🤔. I find it a bit confusing because my brain always associates `exist` to a `test of presence`.

How about changing it to `ensure_exist!`? I find it better express the meaning on what this function does. What do you think?

Reply:
That was the original name!

Then I considered that the exclamation mark at the end means "ensure" which meant that "ensure_exist!" meant "ensure exist ensure". I realized that "exist?" ends with a question mark anyway and the exclamation mark version would pair with it nice.

Reply:
The `!` at the end of the method name doesn't mean "ensure". It means the method has side effects or is destructive.

Reply:
In my mind, the destructive character is synonym to ensure. I renamed the method back.